src/main/java/domain/Attends.java

```diff
@@ -0,0 +1,79 @@
+package domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Attends {
+
+    public List<Attend> attends;
+
+    public Attends(List<Attend> attends) {
+        this.attends = attends;
+    }
+
+    public void addAttend(Attend attend) {
+        validateDuplicate(attend);
+        attends.add(attend);
+    }
+
+    private void validateDuplicate(Attend targetAttend) {
+        if (hasDayEqualsAttend(targetAttend)) {
+            throw new IllegalArgumentException("같은 날짜에 출석할 수 없다.");
+        }
+    }
+
+    public Attend findByDay(int day) {
+        return attends.stream()
+                .filter(attend -> attend.isDayEqual(day))
+                .findFirst()
+                .orElse(Attend.fromDay(day));
+        // TODO: 예외를 던지던 테스트 코드 수정
+    }
+
+    public boolean hasDayEqualsAttend(Attend target) {
+        return attends.stream()
+                .anyMatch(target::isDayEqual);
+    }
+
+    public void edit(Attend attend) {
+        if (hasDayEqualsAttend(attend)) {
+            // 똑같은 거 찾기
+            Attend before = findByAttend(attend);
+
+            // 찾은 객체를 토대로 list에서 삭제
+            removeByAttend(before);
+        }
+        addAttend(attend);
```

Comment:
값이 존재하든 안 하든 remove 호출하고 add하면 결과는 똑같지 않나요? (없으면 remove 호출해도 아무 일도 안 일어나니..)
```suggestion
        removeByAttend(before);
        addAttend(attend);
```

Reply:
그 당시 removeByAttend 메서드를 작성하고 그 다음 edit 메서드를 작성 했을텐데요... 
정말 방금 작성한 코드인데 removeByAttend가 날짜와 시간이 동일한 Attend 객체를 필터링 한다고 생각했습니다.
그래서 기존 Attend 객체를 찾아오는 코드를 작성했습니다. 🙀

리팩터링으로 불필요한 로직인 Attend를 찾아오는 과정을 지우고, removeByAttend 메서드를 동작을 더 잘 설명할 수 있는 removeByDayEqualAttend 이름으로 변경했습니다.
refector(Attends): 로직 상 필요 없는 코드를 제거하도록 수정

더 나은 코드를 위한 제안 감사합니다.

Reply:
`removeByDayEqualAttend`로 변경된 메서드명을 보니 차라리 이름을 `removeByDay`로 변경하고 Attend 대신 `int day`를 받는 것도 방법일 것 같네요ㅎㅎ